x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/SnapshotLifecycleTask.java

```diff
@@ -92,82 +99,108 @@ public static Optional<String> maybeTakeSnapshot(
         Optional<SnapshotLifecyclePolicyMetadata> maybeMetadata = getSnapPolicyMetadata(jobId, clusterService.state());
         String snapshotName = maybeMetadata.map(policyMetadata -> {
             // don't time out on this request to not produce failed SLM runs in case of a temporarily slow master node
-            CreateSnapshotRequest request = policyMetadata.getPolicy().toRequest(TimeValue.MAX_VALUE);
-            final LifecyclePolicySecurityClient clientWithHeaders = new LifecyclePolicySecurityClient(
-                client,
-                ClientHelper.INDEX_LIFECYCLE_ORIGIN,
-                policyMetadata.getHeaders()
-            );
-            logger.info(
-                "snapshot lifecycle policy [{}] issuing create snapshot [{}]",
-                policyMetadata.getPolicy().getId(),
-                request.snapshot()
-            );
-            clientWithHeaders.admin().cluster().createSnapshot(request, new ActionListener<>() {
-                @Override
-                public void onResponse(CreateSnapshotResponse createSnapshotResponse) {
-                    logger.debug(
-                        "snapshot response for [{}]: {}",
-                        policyMetadata.getPolicy().getId(),
-                        Strings.toString(createSnapshotResponse)
-                    );
-                    final SnapshotInfo snapInfo = createSnapshotResponse.getSnapshotInfo();
-                    // Check that there are no failed shards, since the request may not entirely
-                    // fail, but may still have failures (such as in the case of an aborted snapshot)
-                    if (snapInfo.failedShards() == 0) {
-                        long snapshotStartTime = snapInfo.startTime();
-                        final long timestamp = Instant.now().toEpochMilli();
-                        submitUnbatchedTask(
-                            clusterService,
-                            "slm-record-success-" + policyMetadata.getPolicy().getId(),
-                            WriteJobStatus.success(policyMetadata.getPolicy().getId(), request.snapshot(), snapshotStartTime, timestamp)
-                        );
-                        historyStore.putAsync(
-                            SnapshotHistoryItem.creationSuccessRecord(timestamp, policyMetadata.getPolicy(), request.snapshot())
+            CreateSnapshotRequest request = policyMetadata.getPolicy().toRequest(TimeValue.MAX_VALUE, UUIDs.randomBase64UUID());
+            final SnapshotId snapshotId = new SnapshotId(request.snapshot(), request.uuid());
+
+            submitUnbatchedTask(
```

Comment:
`submitUnbatchedTask` is deprecated and should be replaced with  `ClusterStateTaskExecutor`. I have not done this as I don't think we actually could do any batching here. As we don't want to/can't batch anything it seems better to explicitly used the un-batched form. Or should I follow the deprecation warning and update both calls to use `ClusterStateTaskExecutor`?